__tests__/app.test.js

```diff
@@ -129,6 +140,56 @@ describe("GET /api", () => {
             expect(msg).toBe("Bad Request");
           });
       });
+
+      describe("GET /api/articles/:article_id/comments", () => {
+        test("GET 200 - responds with an array of comments for the given article_id of which each comment should have the relevant properties", () => {
+          return request(app)
+            .get(`/api/articles/1/comments`)
+            .expect(200)
+            .then(({ body: { comments } }) => {
+              expect(Array.isArray(comments)).toBe(true);
+              expect(comments.length).toBe(11);
+              comments.forEach((comment) => {
+                expect(comment).toMatchObject({
+                  comment_id: expect.any(Number),
+                  votes: expect.any(Number),
+                  created_at: expect.any(String),
+                  author: expect.any(String),
+                  body: expect.any(String),
+                  article_id: 1,
+                });
+              });
+            });
+        });
+        test("GET 200 - responds with the most recent comments first", () => {
+          return request(app)
+            .get(`/api/articles/1/comments`)
+            .expect(200)
+            .then(({ body: { comments } }) => {
+              expect(comments).toBeSortedBy("created_at", {
+                descending: true,
+              });
+            });
+        });
+        test("GET 404 -- responds with approriate status and message, when the request has invalid article id number", () => {
+          const articleId = 999;
+          return request(app)
+            .get(`/api/articles/${articleId}/comments`)
+            .expect(404)
+            .then(({ body: { msg } }) => {
+              expect(msg).toBe("Not Found");
+            });
+        });
+        test("GET 400 -- responds with approriate status and message, when the request has invalid article id type", () => {
+          const articleId = "papers";
+          return request(app)
+            .get(`/api/articles/${articleId}/comments`)
+            .expect(400)
+            .then(({ body: { msg } }) => {
+              expect(msg).toBe("Bad Request");
+            });
+        });
+      });
```

Comment:
you need a test for an article that has no comments. This should return an empty array with a status code of 200